src/control/UndoRedoController.h

```diff
@@ -44,6 +46,11 @@ class UndoRedoController {
      */
     Layer* layer = nullptr;
 
+    /**
+     * Page of the selection before change
+     */
+    PageRef sourcePage{};
+
```

Comment:
This does not belong here (separation of concerns). Better call `EditSelection* selection = control->getWindow()->getXournal()->getSelection(); ` wherever this is required.

Reply:
As far as I could tell getSelection returns a nullpointer in UndoRedoController::after even when not clearing the selection.